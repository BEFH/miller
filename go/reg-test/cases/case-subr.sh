# Check for BNF/AST errors, with minimal CST involvement
run_mlr -n put -v 'call s()'
run_mlr -n put -v 'call s(1,2,3)'
run_mlr -n put -v 'subr s() {}'
run_mlr -n put -v 'subr s() {x=1}'
run_mlr -n put -v 'subr s() {return}'

# Check for CST invovlement
