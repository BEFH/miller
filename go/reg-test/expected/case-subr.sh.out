
mlr -n put -v call s()
DSL EXPRESSION:
call s()
RAW AST:
* statement block
    * subroutine callsite "call"


mlr -n put -v call s(1,2,3)
DSL EXPRESSION:
call s(1,2,3)
RAW AST:
* statement block
    * subroutine callsite "call"
        * int literal "1"
        * int literal "2"
        * int literal "3"


mlr -n put -v subr s() {}
DSL EXPRESSION:
subr s() {}
RAW AST:
* statement block
    * subroutine definition "s"
        * parameter list
        * statement block


mlr -n put -v subr s() {x=1}
DSL EXPRESSION:
subr s() {x=1}
RAW AST:
* statement block
    * subroutine definition "s"
        * parameter list
        * statement block
            * assignment "="
                * local variable "x"
                * int literal "1"


mlr -n put -v subr s() {return}
DSL EXPRESSION:
subr s() {return}
RAW AST:
* statement block
    * subroutine definition "s"
        * parameter list
        * statement block
            * return "return"


